Makefile

```diff
@@ -1,61 +1,103 @@
 SHELL=/bin/bash
+ELASTIC_REGISTRY := docker.elastic.co
+
 export PATH := ./bin:./venv/bin:$(PATH)
 
 ifndef ELASTIC_VERSION
-ELASTIC_VERSION := $(shell ./bin/elastic-version)
+  ELASTIC_VERSION := $(shell ./bin/elastic-version)
 endif
 
 ifdef STAGING_BUILD_NUM
-VERSION_TAG := $(ELASTIC_VERSION)-$(STAGING_BUILD_NUM)
+  VERSION_TAG := $(ELASTIC_VERSION)-$(STAGING_BUILD_NUM)
 else
-VERSION_TAG := $(ELASTIC_VERSION)
+  VERSION_TAG := $(ELASTIC_VERSION)
 endif
 
-ELASTIC_REGISTRY := docker.elastic.co
-VERSIONED_IMAGE := $(ELASTIC_REGISTRY)/elasticsearch/elasticsearch:$(VERSION_TAG)
+# Build different images tagged as :version-<flavor>
+ifndef IMAGE_FLAVORS
+  # basic license not available as of 6.0.0-beta1
+  # IMAGE_FLAVORS := oss basic platinum
+  IMAGE_FLAVORS := oss platinum
+endif
+
+# Which image flavor will additionally receive the plain `:version` tag
+ifndef DEFAULT_IMAGE_FLAVOR
+  DEFAULT_IMAGE_FLAVOR := oss
+endif
+
+ifndef VERSIONED_IMAGE
+  VERSIONED_IMAGE := $(ELASTIC_REGISTRY)/elasticsearch/elasticsearch:$(VERSION_TAG)
+endif
 
 # When invoking docker-compose, use an extra config fragment to map Elasticsearch's
 # listening port to the docker host.
-DOCKER_COMPOSE := docker-compose -f docker-compose.yml -f docker-compose.hostports.yml
+# For the x-pack security enabled image (platinum), use the fragment we utilize for tests.
+ifeq ($(DEFAULT_IMAGE_FLAVOR),platinum)
+  DOCKER_COMPOSE := docker-compose \
+	-f docker-compose-$(DEFAULT_IMAGE_FLAVOR).yml \
+	-f tests/docker-compose-$(DEFAULT_IMAGE_FLAVOR).yml
+else
+  DOCKER_COMPOSE := docker-compose \
+	-f docker-compose-$(DEFAULT_IMAGE_FLAVOR).yml \
+	-f docker-compose.hostports.yml
+endif
 
-.PHONY: all test lint clean pristine run run-single run-cluster build release-manager-snapshot push
+.PHONY: all dockerfile docker-compose test test-only lint clean pristine run run-single run-cluster build release-manager-snapshot push
 
 # Default target, build *and* run tests
 all: build test
 
-test: lint build docker-compose.yml
-	./bin/pytest tests
-	./bin/pytest --single-node tests
+test: lint build docker-compose test-only
+
+# Test specified versions without building
+test-only: lint docker-compose
+	$(foreach FLAVOR, $(IMAGE_FLAVORS), \
+	pyfiglet -w 160 -f puffy "test: $(FLAVOR) image"; \
```

Comment:
Adorable _and_ useful. 💚

Why it's not called "Pyglet" remains a mystery for the ages.